warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:38:9
   |
LL | #![warn(x5400)] //~ WARN unknown lint: `x5400`
   |         ^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:33:28
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |                            ^^^^^^^^^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:39:10
   |
LL | #![allow(x5300)] //~ WARN unknown lint: `x5300`
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:40:11
   |
LL | #![forbid(x5200)] //~ WARN unknown lint: `x5200`
   |           ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:41:9
   |
LL | #![deny(x5100)] //~ WARN unknown lint: `x5100`
   |         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:102:8
   |
LL | #[warn(x5400)]
   |        ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:105:25
   |
LL |     mod inner { #![warn(x5400)] }
   |                         ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:108:12
   |
LL |     #[warn(x5400)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:111:12
   |
LL |     #[warn(x5400)] struct S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:114:12
   |
LL |     #[warn(x5400)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5400`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:117:12
   |
LL |     #[warn(x5400)] impl S { }
   |            ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:121:9
   |
LL | #[allow(x5300)]
   |         ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:124:26
   |
LL |     mod inner { #![allow(x5300)] }
   |                          ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:127:13
   |
LL |     #[allow(x5300)] fn f() { }
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:130:13
   |
LL |     #[allow(x5300)] struct S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:133:13
   |
LL |     #[allow(x5300)] type T = S;
   |             ^^^^^

warning: unknown lint: `x5300`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:136:13
   |
LL |     #[allow(x5300)] impl S { }
   |             ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:140:10
   |
LL | #[forbid(x5200)]
   |          ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:143:27
   |
LL |     mod inner { #![forbid(x5200)] }
   |                           ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:146:14
   |
LL |     #[forbid(x5200)] fn f() { }
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:149:14
   |
LL |     #[forbid(x5200)] struct S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:152:14
   |
LL |     #[forbid(x5200)] type T = S;
   |              ^^^^^

warning: unknown lint: `x5200`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:155:14
   |
LL |     #[forbid(x5200)] impl S { }
   |              ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:159:8
   |
LL | #[deny(x5100)]
   |        ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:162:25
   |
LL |     mod inner { #![deny(x5100)] }
   |                         ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:165:12
   |
LL |     #[deny(x5100)] fn f() { }
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:168:12
   |
LL |     #[deny(x5100)] struct S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:171:12
   |
LL |     #[deny(x5100)] type T = S;
   |            ^^^^^

warning: unknown lint: `x5100`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:174:12
   |
LL |     #[deny(x5100)] impl S { }
   |            ^^^^^

warning: `macro_escape` is a deprecated synonym for `macro_use`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:458:1
   |
LL | #[macro_escape]
   | ^^^^^^^^^^^^^^^

warning: `macro_escape` is a deprecated synonym for `macro_use`
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:461:17
   |
LL |     mod inner { #![macro_escape] }
   |                 ^^^^^^^^^^^^^^^^
   |
   = help: consider an outer attribute, `#[macro_use] mod ...`

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:182:5
   |
LL |     #[macro_use] fn f() { }
   |     ^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:33:9
   |
LL | #![warn(unused_attributes, unknown_lints)]
   |         ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:185:5
   |
LL |     #[macro_use] struct S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:188:5
   |
LL |     #[macro_use] type T = S;
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:191:5
   |
LL |     #[macro_use] impl S { }
   |     ^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:198:17
   |
LL |     mod inner { #![macro_export] }
   |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:201:5
   |
LL |     #[macro_export] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:204:5
   |
LL |     #[macro_export] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:207:5
   |
LL |     #[macro_export] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:210:5
   |
LL |     #[macro_export] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:195:1
   |
LL | #[macro_export]
   | ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:217:17
   |
LL |     mod inner { #![plugin_registrar] }
   |                 ^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:222:5
   |
LL |     #[plugin_registrar] struct S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:225:5
   |
LL |     #[plugin_registrar] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:228:5
   |
LL |     #[plugin_registrar] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:214:1
   |
LL | #[plugin_registrar]
   | ^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:235:17
   |
LL |     mod inner { #![main] }
   |                 ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:240:5
   |
LL |     #[main] struct S;
   |     ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:243:5
   |
LL |     #[main] type T = S;
   |     ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:246:5
   |
LL |     #[main] impl S { }
   |     ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:232:1
   |
LL | #[main]
   | ^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:253:17
   |
LL |     mod inner { #![start] }
   |                 ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:258:5
   |
LL |     #[start] struct S;
   |     ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:261:5
   |
LL |     #[start] type T = S;
   |     ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:264:5
   |
LL |     #[start] impl S { }
   |     ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:250:1
   |
LL | #[start]
   | ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:317:5
   |
LL |     #[path = "3800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:320:5
   |
LL |     #[path = "3800"]  struct S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:323:5
   |
LL |     #[path = "3800"] type T = S;
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:326:5
   |
LL |     #[path = "3800"] impl S { }
   |     ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:333:17
   |
LL |     mod inner { #![automatically_derived] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:336:5
   |
LL |     #[automatically_derived] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:339:5
   |
LL |     #[automatically_derived] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:342:5
   |
LL |     #[automatically_derived] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:345:5
   |
LL |     #[automatically_derived] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:330:1
   |
LL | #[automatically_derived]
   | ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:365:17
   |
LL |     mod inner { #![no_link] }
   |                 ^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:368:5
   |
LL |     #[no_link] fn f() { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:371:5
   |
LL |     #[no_link] struct S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:374:5
   |
LL |     #[no_link]type T = S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:377:5
   |
LL |     #[no_link] impl S { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:362:1
   |
LL | #[no_link]
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:384:17
   |
LL |     mod inner { #![should_panic] }
   |                 ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:387:5
   |
LL |     #[should_panic] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:390:5
   |
LL |     #[should_panic] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:393:5
   |
LL |     #[should_panic] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:396:5
   |
LL |     #[should_panic] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:381:1
   |
LL | #[should_panic]
   | ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:403:17
   |
LL |     mod inner { #![ignore] }
   |                 ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:406:5
   |
LL |     #[ignore] fn f() { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:409:5
   |
LL |     #[ignore] struct S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:412:5
   |
LL |     #[ignore] type T = S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:415:5
   |
LL |     #[ignore] impl S { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:400:1
   |
LL | #[ignore]
   | ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:422:17
   |
LL |     mod inner { #![no_implicit_prelude] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:425:5
   |
LL |     #[no_implicit_prelude] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:428:5
   |
LL |     #[no_implicit_prelude] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:431:5
   |
LL |     #[no_implicit_prelude] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:434:5
   |
LL |     #[no_implicit_prelude] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:419:1
   |
LL | #[no_implicit_prelude]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:441:17
   |
LL |     mod inner { #![reexport_test_harness_main="2900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:444:5
   |
LL |     #[reexport_test_harness_main = "2900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:447:5
   |
LL |     #[reexport_test_harness_main = "2900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:450:5
   |
LL |     #[reexport_test_harness_main = "2900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:453:5
   |
LL |     #[reexport_test_harness_main = "2900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:438:1
   |
LL | #[reexport_test_harness_main = "2900"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:464:5
   |
LL |     #[macro_escape] fn f() { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:467:5
   |
LL |     #[macro_escape] struct S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:470:5
   |
LL |     #[macro_escape] type T = S;
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:473:5
   |
LL |     #[macro_escape] impl S { }
   |     ^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:481:17
   |
LL |     mod inner { #![no_std] }
   |                 ^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:481:17
   |
LL |     mod inner { #![no_std] }
   |                 ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:485:5
   |
LL |     #[no_std] fn f() { }
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:485:5
   |
LL |     #[no_std] fn f() { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:489:5
   |
LL |     #[no_std] struct S;
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:489:5
   |
LL |     #[no_std] struct S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:493:5
   |
LL |     #[no_std] type T = S;
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:493:5
   |
LL |     #[no_std] type T = S;
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:497:5
   |
LL |     #[no_std] impl S { }
   |     ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:497:5
   |
LL |     #[no_std] impl S { }
   |     ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:477:1
   |
LL | #[no_std]
   | ^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:477:1
   |
LL | #[no_std]
   | ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:636:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:636:17
   |
LL |     mod inner { #![crate_name="0900"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:640:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:640:5
   |
LL |     #[crate_name = "0900"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:644:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:644:5
   |
LL |     #[crate_name = "0900"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:648:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:648:5
   |
LL |     #[crate_name = "0900"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:652:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:652:5
   |
LL |     #[crate_name = "0900"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:632:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:632:1
   |
LL | #[crate_name = "0900"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:661:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:661:17
   |
LL |     mod inner { #![crate_type="0800"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:665:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:665:5
   |
LL |     #[crate_type = "0800"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:669:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:669:5
   |
LL |     #[crate_type = "0800"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:673:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:673:5
   |
LL |     #[crate_type = "0800"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:677:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:677:5
   |
LL |     #[crate_type = "0800"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:657:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:657:1
   |
LL | #[crate_type = "0800"]
   | ^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:686:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:686:17
   |
LL |     mod inner { #![feature(x0600)] }
   |                 ^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:690:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:690:5
   |
LL |     #[feature(x0600)] fn f() { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:694:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:694:5
   |
LL |     #[feature(x0600)] struct S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:698:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:698:5
   |
LL |     #[feature(x0600)] type T = S;
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:702:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:702:5
   |
LL |     #[feature(x0600)] impl S { }
   |     ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:682:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:682:1
   |
LL | #[feature(x0600)]
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:712:17
   |
LL |     mod inner { #![no_main] }
   |                 ^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:712:17
   |
LL |     mod inner { #![no_main] }
   |                 ^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:716:5
   |
LL |     #[no_main] fn f() { }
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:716:5
   |
LL |     #[no_main] fn f() { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:720:5
   |
LL |     #[no_main] struct S;
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:720:5
   |
LL |     #[no_main] struct S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:724:5
   |
LL |     #[no_main] type T = S;
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:724:5
   |
LL |     #[no_main] type T = S;
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:728:5
   |
LL |     #[no_main] impl S { }
   |     ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:728:5
   |
LL |     #[no_main] impl S { }
   |     ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:708:1
   |
LL | #[no_main]
   | ^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:708:1
   |
LL | #[no_main]
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:750:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:750:17
   |
LL |     mod inner { #![recursion_limit="0200"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:754:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:754:5
   |
LL |     #[recursion_limit="0200"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:758:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:758:5
   |
LL |     #[recursion_limit="0200"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:762:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:762:5
   |
LL |     #[recursion_limit="0200"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:766:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:766:5
   |
LL |     #[recursion_limit="0200"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:746:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:746:1
   |
LL | #[recursion_limit="0200"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:775:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be in the root module
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:775:17
   |
LL |     mod inner { #![type_length_limit="0100"] }
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:779:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:779:5
   |
LL |     #[type_length_limit="0100"] fn f() { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:783:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:783:5
   |
LL |     #[type_length_limit="0100"] struct S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:787:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:787:5
   |
LL |     #[type_length_limit="0100"] type T = S;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:791:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:791:5
   |
LL |     #[type_length_limit="0100"] impl S { }
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:771:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: crate-level attribute should be an inner attribute: add an exclamation mark: #![foo]
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:771:1
   |
LL | #[type_length_limit="0100"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:43:1
   |
LL | #![macro_export] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:44:1
   |
LL | #![plugin_registrar] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:47:1
   |
LL | #![main] //~ WARN unused attribute
   | ^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:48:1
   |
LL | #![start] //~ WARN unused attribute
   | ^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:51:1
   |
LL | #![repr()]
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:53:1
   |
LL | #![path = "3800"] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:54:1
   |
LL | #![automatically_derived] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:56:1
   |
LL | #![no_link] //~ WARN unused attribute
   | ^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:58:1
   |
LL | #![should_panic] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:59:1
   |
LL | #![ignore] //~ WARN unused attribute
   | ^^^^^^^^^^

warning: unused attribute
  --> $DIR/issue-43106-gating-of-builtin-attrs.rs:66:1
   |
LL | #![proc_macro_derive()] //~ WARN unused attribute
   | ^^^^^^^^^^^^^^^^^^^^^^^

error: invalid windows subsystem `1000`, only `windows` and `console` are allowed

error: aborting due to previous error

